hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/BucketHandler.java

```diff
@@ -196,8 +197,12 @@ public static BucketHandler getBucketHandler(
       String volumeName, String bucketName) throws IOException {
 
     String bucketKey = omMetadataManager.getBucketKey(volumeName, bucketName);
-    OmBucketInfo bucketInfo = omMetadataManager
-        .getBucketTable().getSkipCache(bucketKey);
+    Table<String, OmBucketInfo> bucketTable =
+        omMetadataManager.getBucketTable();
+    OmBucketInfo bucketInfo = null;
+    if (null != bucketTable) {
+      bucketInfo = bucketTable.getSkipCache(bucketKey);
+    }
 
     return getBucketHandler(reconNamespaceSummaryManager,
         omMetadataManager, reconSCM, bucketInfo);
```

Comment:
We can move getBucketHandler call also inside if (null != bucketTable), so that no need to call getBucketHandler if bucketInfo is null.

Reply:
> We can move getBucketHandler call also inside if (null != bucketTable), so that no need to call getBucketHandler if bucketInfo is null.

Thanks Ashish for review, however this is not needed as already below check is available in existing code of `getBucketHandler:`
![Uploading image.png…]()